vllm/entrypoints/openai/api_server.py

```diff
@@ -821,6 +854,13 @@ async def init_app_state(
         chat_template=resolved_chat_template,
         chat_template_content_format=args.chat_template_content_format,
     )
+    state.openai_serving_transcription = OpenAIServingTranscription(
+        engine_client,
+        model_config,
+        state.openai_serving_models,
+        request_logger=request_logger,
+    ) if ("WhisperForConditionalGeneration"
+          in model_config.hf_config.architectures) else None
```

Comment:
Since we aren't relying on a task for this, I assume we are defaulting to the "generate" task. What happens if the user tries to use the OpenAIServingCompletion api also available in this case?

Reply:
Great point, we should probably make this endpoint mutually exclusive with a any other that checks for 'generate'. Or do you have a better idea in mind..?

Reply:
Yeah I think we just don't want to expose endpoints that we know will not work. You could either do that by making a new task that whisper maps too i.e. 'transcription' or do this special-case in-place like you have here.

Reply:
The new task solution looks cleaner

Reply:
New logs shoudl look like this : 
`This model supports multiple tasks: {'score', 'embed', 'reward', 'generate', 'transcription', 'classify'}. Defaulting to 'transcription'.`